pkg/plugin/driver/base_test.go

```diff
@@ -101,3 +105,161 @@ func TestSkipCleanup(t *testing.T) {
+		}
+
+		for i, c := range container.Command {
+			if c != command[i] {
+				return fmt.Errorf("expected command item %v to be %q, got %q", i, command[i], c)
+			}
+		}
+
+		for i, arg := range container.Args {
+			if arg != args[i] {
+				return fmt.Errorf("expected args item %v to be %q, got %q", i, args[i], arg)
+			}
+		}
+		return nil
+	}
+
+	checkEnvironment := func(container v1.Container) error {
+		expectedEnvVars := []v1.EnvVar{
+			{
+				Name:  "RESULTS_DIR",
+				Value: "/tmp/results",
+			},
+			{
+				Name:  "RESULT_TYPE",
+				Value: b.GetResultType(),
+			},
+			{
+				Name:  "MASTER_URL",
+				Value: aggregatorURL,
+			},
+			{
+				Name:  "CA_CERT",
+				Value: "",
+			},
+		}
+		for _, e := range expectedEnvVars {
+			if !envContains(container.Env, e) {
+				return fmt.Errorf("expected container environment to contain %q", e)
+			}
+		}
+		for _, e := range container.Env {
+			switch e.Name {
+			case "NODE_NAME":
+				expected := "spec.nodeName"
+				got := e.ValueFrom.FieldRef.FieldPath
+				if got != expected {
+					return fmt.Errorf("expected NODE_NAME to have FieldRef value %q, but got %q", expected, got)
+				}
+			case "CLIENT_CERT":
+				expectedKey := "tls.crt"
+				gotKey := e.ValueFrom.SecretKeyRef.Key
+				if gotKey != expectedKey {
+					return fmt.Errorf("expected CLIENT_CERT to have SecretKeyRef key %q, but got %q", expectedKey, gotKey)
+				}
+				expectedName := "sonobuoy-plugin-test-plugin-sessionID"
+				gotName := e.ValueFrom.SecretKeyRef.LocalObjectReference.Name
+				if gotKey != expectedKey {
+					return fmt.Errorf("expected CLIENT_CERT to have SecretKeyRef name %q, but got %q", expectedName, gotName)
+				}
+			case "CLIENT_KEY":
+				expectedKey := "tls.key"
+				gotKey := e.ValueFrom.SecretKeyRef.Key
+				if gotKey != expectedKey {
+					return fmt.Errorf("expected CLIENT_CERT to have SecretKeyRef key %q, but got %q", expectedKey, gotKey)
+				}
+				expectedName := "sonobuoy-plugin-test-plugin-sessionID"
+				gotName := e.ValueFrom.SecretKeyRef.LocalObjectReference.Name
+				if gotKey != expectedKey {
+					return fmt.Errorf("expected CLIENT_CERT to have SecretKeyRef name %q, but got %q", expectedName, gotName)
+				}
+			}
+		}
+		return nil
+	}
+
+	checkVolumes := func(container v1.Container) error {
+		expectedVolume := v1.VolumeMount{
+			Name:      "results",
+			ReadOnly:  false,
+			MountPath: "/tmp/results",
+		}
+		if len(container.VolumeMounts) != 1 {
+			return fmt.Errorf("expected the container to have one volume mount, got %v", len(container.VolumeMounts))
+		}
+		if container.VolumeMounts[0] != expectedVolume {
+			return fmt.Errorf("expected volume mount to equal %v, got %v", expectedVolume, container.VolumeMounts[0])
+		}
+		return nil
+
+	}
+
+	testCases := []struct {
+		desc  string
+		check func(v1.Container) error
+	}{
+		{
+			"Fields set correctly",
+			checkFields,
```

Comment:
👍 I do like reading the test more this way. It is also clear how I could/should expand the functionality if I needed to test something else.